pkg/controller/gitopscluster/gitopscluster_controller.go

```diff
@@ -656,14 +671,31 @@ func (r *ReconcileGitOpsCluster) GetManagedClusters(namespace string, placementr
 		return nil, errors.New("no placement decision found for placement: " + placementref.Name)
 	}
 
-	clusters := make([]string, 0)
+	clusterList := &spokeclusterv1.ManagedClusterList{}
+
+	err = r.List(context.TODO(), clusterList, &client.ListOptions{})
+	if err != nil {
+		klog.Error("Failed to list managed clusters, err:", err)
+		return nil, err
+	}
+
+	clusters := make([]spokeclusterv1.ManagedCluster, 0)
 
 	for _, placementdecision := range placementDecisions.Items {
 		klog.Info("getting cluster names from placement decision " + placementdecision.Name)
 
 		for _, clusterDecision := range placementdecision.Status.Decisions {
 			klog.Info("cluster name: " + clusterDecision.ClusterName)
-			clusters = append(clusters, clusterDecision.ClusterName)
+
+			for i, cluster := range clusterList.Items {
+				if cluster.Name == clusterDecision.ClusterName {
+					klog.Info("found managed cluster: " + clusterDecision.ClusterName)
+
+					clusters = append(clusters, clusterList.Items[i])
+
+					break
+				}
+			}
 		}
```

Comment:
There could be a performance issue. e.g.  if the whole number of managed cluster is 3K (that is the scenario that ACM supports now),  for each matched cluster name in the placementDecision list, it needs 3k loop to find its managedCluster.

I would suggest to save all managed Cluster manifest to a map for later faster search/match, where the cluster name is the key of the map so sth like this:

err = r.List(context.TODO(), clusterList, &client.ListOptions{})
if err != nil {
    klog.Error("Failed to list managed clusters, err:", err)
    return nil, err
}

clusterMap := make(map[string]spokeclusterv1.ManagedCluster)
for _, managedCluster := range clusterList.Items {
    clustermap[managedCluster.Name] =  managedCluster
}

Then in the later placementDecision loop

for _, clusterDecision := range placementdecision.Status.Decisions {
    if cluster, ok :=  clustermap[clusterDecision.ClusterName] {
        clusters = append(clusters, cluster)
    }
}

Reply:
Very good suggestion, I wasn't content with this implementation too. I updated it to follow your suggestion.